### ISSUE TITLE
Static extensions with generic type parameters 

### ISSUE THREAD

**Issue by markknol** [HaxeFoundation/haxe#8760]
Kinda hard to describe but here we go
- If there is a module with one class with a parameterized function designed for static extension then everything is fine 
- If there is a module with two different classes with both a parameterized function designed for static extension there is a this kind of parameter type `<T:Float>`  it works
- Problem: If there is a module with two different classes with both a parameterized function designed for static extension, but there is this kind of parameter type  `<T:Float, A:Array<T>>`  then I get a `"Constraint check failure for sorted.T"` error. 

This case works
```haxe
using Test;

class Test {
    static function main() {
        var arr2 = [123,456];
        trace(arr2.sorted());
    }
}

class SortFloatTools {
  public static function sorted<T:Float, A:Array<T>>(arr:A):A {
      trace('sorted<T:Float>()');
      return arr;
  }
}
```
This case works
```haxe
using Test;

class Test {
    static function main() {
        trace("test".func());
        trace(123.func());
    }
}

class SortStringTools {
  public static function func<T:String>(v:T):T {
      trace('sorted<T:String>()');
      return v;
  }
}
class SortFloatTools {
  public static function func<T:Float>(v:T):T {
      trace('sorted<T:Float>()');
      return v;
  }
}
```
This case *does not* compile:
```haxe
using Test;

class Test {
    static function main() {
        var arr1 = ["abc", "def"];
        trace(arr1.sorted());
        
        var arr2 = [123,456];
        trace(arr2.sorted());
    }
}

class SortStringTools {
  public static function sorted<T:String, A:Array<T>>(arr:A):A {
      trace('sorted<T:String>()');
      return arr;
  }
}
class SortFloatTools {
  public static function sorted<T:Float, A:Array<T>>(arr:A):A {
      trace('sorted<T:Float>()');
      return arr;
  }
}
```

Seen in both Haxe 3 and Haxe 4 rc3

**Comment by Simn** [comment]
This should be a one-line fix after merging #9549.

**Comment by markknol** [comment]
Nice! This is a great feature!
